src/app/pages/courses-page/components/courses-list/courses-list.component.ts

```diff
@@ -8,30 +7,40 @@ import Course from '../course/course.types';
   templateUrl: './courses-list.component.html',
   styleUrls: ['./courses-list.component.scss'],
 })
-export class CoursesListComponent implements DoCheck{
-  @Input() searchCourse: string;
-
-  private courses: Course[];
+export class CoursesListComponent implements OnInit, OnChanges {
+  start = 0;
+  count = 5;
+  sort = 'date';
+  @Input() textFragment = '';
+  coursesList: Array<Course> = [];
 
   constructor(
-    private filterPipe: FilterPipe,
-    private coursesService: CoursesService,
-    ) {}
+    public coursesService: CoursesService,
+  ) {
+    this.coursesService.coursesChanged.subscribe(() => this.loadData());
```

Comment:
Don't use constructor for defining some value or something else. Best practice is to use constructor for DI. Define value for class properties in lifecycle hooks or declaration space.

Reply:
I moved the subscription to the ngOnInit method.